### ISSUE TITLE
Error Parsing SFTDataset

### ISSUE THREAD

**Issue by TJ-Solergibert** [axolotl-ai-cloud/axolotl#1500]
### Please check that this issue hasn't been reported before.

- [X] I searched previous [Bug Reports](https://github.com/OpenAccess-AI-Collective/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

After parsing the .yml file, I should get a `SFTDataset` with the `conversation` and roles field. 

### Current behaviour

I get a `DPODataset` with the `path`, `type` and `data_files = None` fields. If I hardcode the conversation field in `axolotl/prompt_strategies/sharegpt.py L43` everything works as expected

### Steps to reproduce

In [`src/axolotl/utils/config/__init__.py` L208](https://github.com/OpenAccess-AI-Collective/axolotl/blob/ff01c45127aee0eb607afe20ea5cde78f67e0791/src/axolotl/utils/config/__init__.py#L208) `AxolotlConfigWCapabilities` parses the datasets field of the config file as a `DPODataset` and not as a `SFTDataset`.

```
debug = AxolotlConfigWCapabilities(**cfg.to_dict(), capabilities=capabilities)
debug.datasets
[DPODataset(path='/mloscratch/homes/solergib/datasets/SlimOrca/oo-labeled_correct.gpt4...in_sft', type='sharegpt', data_files=None)]
```

### Config yaml

```yaml
base_model: mistralai/Mistral-7B-v0.1
model_type: MistralForCausalLM
tokenizer_type: LlamaTokenizer

load_in_8bit: false
load_in_4bit: false
strict: false

datasets:
  - path: /mloscratch/homes/solergib/datasets/SlimOrca/oo-labeled_correct.gpt4.sharegpt.jsonl
    conversation: mistral # Options (see Conversation 'name'): https://github.com/lm-sys/FastChat/blob/main/fastchat/conversation.py
    split: train_sft
    type: sharegpt
    field_human: human # Optional[str]. Human key to use for conversation.
    field_model: gpt # Optional[str]. Assistant key to use for conversation.
    # Add additional keys from your dataset as input or output roles
    roles:
      input: human,system # Optional[List[str]]. These will be masked based on train_on_input
      output: gpt # Optional[List[str]].
dataset_prepared_path:
val_set_size: 0.05
output_dir: ./out
dataset_processes: 64


sequence_len: 8192
sample_packing: true
pad_to_sequence_len: true
eval_sample_packing: false

wandb_project:
wandb_entity:
wandb_watch:
wandb_name:
wandb_log_model:

max_steps: 16
#batch_size: 4 
gradient_accumulation_steps: 4
micro_batch_size: 2
num_epochs: 4
optimizer: adamw_bnb_8bit
lr_scheduler: cosine
learning_rate: 0.000005

train_on_inputs: false
group_by_length: false
bf16: auto
fp16:
tf32: false

gradient_checkpointing: true
early_stopping_patience:
resume_from_checkpoint:
local_rank:
logging_steps: 1
xformers_attention:
flash_attention: true

warmup_steps: 10
evals_per_epoch: 4
eval_table_size:
eval_max_new_tokens: 128
saves_per_epoch: 1
debug:
deepspeed:
weight_decay: 0.0
fsdp:
fsdp_config:
special_tokens:
```

### Which Operating Systems are you using?

- [X] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.10.12

### axolotl branch-commit

main/2fa65b9

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this bug has not been reported yet.
- [X] I am using the latest version of axolotl.
- [X] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by NanoCode012** [comment]
Do you get an error during train to validate this error?

**Comment by TJ-Solergibert** [comment]
As it's not parsing correctly the config, it doesn't applies correctly the `mistral` `conversation` and so on...

**Comment by NanoCode012** [comment]
Can you run https://github.com/OpenAccess-AI-Collective/axolotl?tab=readme-ov-file#preprocess-dataset with --debug

**Comment by TJ-Solergibert** [comment]
Issue solved, next time I should pay more attention to typing. 

```
roles:
      input: human,system # Optional[List[str]]. These will be masked based on train_on_input
      output: gpt # Optional[List[str]].
```
**List[str]** 😅

Thanks!
